integration/cluster_test.go

```diff
@@ -371,3 +372,38 @@ func clusterMustProgress(t *testing.T, membs []*member) {
 		mcancel()
 	}
 }
+
+func TestStopTriggerTransferLeader(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := NewClusterV3(t, &ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	leaderIdx1 := clus.waitLeader(t, clus.Members)
+	leaderID1 := clus.Members[leaderIdx1].s.ID()
+
+	donec := make(chan struct{})
+	go func() {
+		defer close(donec)
+
+		// fail when the leader is taken down without transfer
+		// and succeed if transfer works
+		kvc := toGRPC(clus.Client((leaderIdx1 + 1) % 3)).KV
+		key := []byte("foo")
+		if _, err := kvc.Range(context.TODO(), &etcdserverpb.RangeRequest{Key: key}); err != nil {
```

Comment:
I think this should be more aggressive by hammering `Range` requests until fetching `leaderID2`. Otherwise the Range could slip through before the etcd server shuts down.